data/scripts/sol-flow-node-type-gen.py.in

```diff
@@ -663,6 +676,33 @@ sol_flow_foreach_builtin_node_type_%(name)s(bool (*cb)(void *data, const struct
 #endif // SOL_FLOW_NODE_TYPE_DESCRIPTION_ENABLED
 """)
 
+def generate_composed_type_get_function(outfile, composed_type, prefix):
+    type_names = composed_type.split(":", 1)[1]
+    type_names_underscore = type_names.replace(",", "_")
+    packet_types = type_names.split(",")
+    packet_type_final = []
+    for packet_type in packet_types:
+        packet_type_final.append(data_type_to_packet_type_map[packet_type])
+    sig = "%s_get_composed_%s_packet_type(void)" % (prefix, type_names_underscore)
+
+    if composed_types_signatures[sig] == True:
+        return
+    composed_types_signatures[sig] = True
+    outfile.write("""
```

Comment:
I think you should postpone this `outfile.write(...)` block and do it at once for all types, then you also take the care to iterate over `sorted(composed_types_signatures.keys())` to make the generated C stable (types always in the same order).